src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/Validation/ValidationStateDictionary.cs

```diff
@@ -0,0 +1,179 @@
+            {
+                return ((IDictionary<object, ValidationStateEntry>)_inner).Keys;
+            }
+        }
+
+        /// <inheritdoc />
+        public ICollection<ValidationStateEntry> Values
+        {
+            get
+            {
+                return ((IDictionary<object, ValidationStateEntry>)_inner).Values;
+            }
+        }
+
+        /// <inheritdoc />
+        IEnumerable<object> IReadOnlyDictionary<object, ValidationStateEntry>.Keys
+        {
+            get
+            {
+                return ((IReadOnlyDictionary<object, ValidationStateEntry>)_inner).Keys;
+            }
+        }
+
+        /// <inheritdoc />
+        IEnumerable<ValidationStateEntry> IReadOnlyDictionary<object, ValidationStateEntry>.Values
+        {
+            get
+            {
+                return ((IReadOnlyDictionary<object, ValidationStateEntry>)_inner).Values;
+            }
+        }
+
+        /// <inheritdoc />
+        public void Add(KeyValuePair<object, ValidationStateEntry> item)
+        {
+            ((IDictionary<object, ValidationStateEntry>)_inner).Add(item);
+        }
+
+        /// <inheritdoc />
+        public void Add(object key, ValidationStateEntry value)
+        {
+            _inner.Add(key, value);
+        }
+
+        /// <inheritdoc />
+        public void Clear()
+        {
+            _inner.Clear();
+        }
+
+        /// <inheritdoc />
+        public bool Contains(KeyValuePair<object, ValidationStateEntry> item)
+        {
+            return ((IDictionary<object, ValidationStateEntry>)_inner).Contains(item);
+        }
+
+        /// <inheritdoc />
+        public bool ContainsKey(object key)
+        {
+            return _inner.ContainsKey(key);
+        }
+
+        /// <inheritdoc />
+        public void CopyTo(KeyValuePair<object, ValidationStateEntry>[] array, int arrayIndex)
+        {
+            ((IDictionary<object, ValidationStateEntry>)_inner).CopyTo(array, arrayIndex);
+        }
+
+        /// <inheritdoc />
+        public IEnumerator<KeyValuePair<object, ValidationStateEntry>> GetEnumerator()
+        {
+            return ((IDictionary<object, ValidationStateEntry>)_inner).GetEnumerator();
+        }
+
+        /// <inheritdoc />
+        public bool Remove(KeyValuePair<object, ValidationStateEntry> item)
+        {
+            return _inner.Remove(item);
+        }
+
+        /// <inheritdoc />
+        public bool Remove(object key)
+        {
+            return _inner.Remove(key);
+        }
+
+        /// <inheritdoc />
+        public bool TryGetValue(object key, out ValidationStateEntry value)
+        {
+            return _inner.TryGetValue(key, out value);
+        }
+
+        /// <inheritdoc />
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return ((IDictionary<object, ValidationStateEntry>)_inner).GetEnumerator();
+        }
+        
+        private class ReferenceEqualityComparer : IEqualityComparer<object>
```

Comment:
Two copies of this code now. Perhaps the version in Mvc.Core should be public and in Mvc.Abstractions?